test/unit/plugins/connection_test.rb

```diff
@@ -78,6 +78,54 @@
          .method(:logger).call.must_equal(l)
     end
 
+    describe 'cached_client helper' do
+      class DemoConnection < Train::Plugins::Transport::BaseConnection
+        def initialize(options = {})
+          super(options)
+          @cache_enabled[:api_call] = true
+          @cache[:api_call] = {}
+        end
+
+        def demo_client
+          cached_client(:api_call, :demo_client) do
+            DemoClient.new
+          end
+        end
+
+        class DemoClient
+        end
+      end
+
+      it 'returns a new connection when cached disabled' do
+        conn = DemoConnection.new
+        conn.disable_cache(:api_call)
+
+        client1 = conn.demo_client
+        client2 = conn.demo_client
+
+        client1.wont_be_same_as client2
+      end
+
+      it 'returns a new connection when cache enabled and not hydrated' do
+        conn = DemoConnection.new
+        conn.enable_cache(:api_call)
+
+        client1 = conn.demo_client
```

Comment:
It wouldn't be useful in the other tests, because they don't look anything like real world code, but this one made me wonder if would just return `self` from `enable_cache()` and `disable_cache()`. I quick search shows that the implicit return of a boolean is never treated as significant.

```ruby
def enable_cache(type)
  fail Train::UnknownCacheType, "#{type} is not a valid cache type" unless @cache_enabled.keys.include?(type.to_sym)

  @cache_enabled[type.to_sym] = true

  self
end
```
```ruby
it 'returns a new connection when cache enabled and not hydrated' do
  conn = DemoConnection.new.enable_cache(:api_call)
￼
  client1 = conn.demo_client
  client1.must_be_instance_of DemoConnection::DemoClient
end

# or even...

it 'returns a new connection when cache enabled and not hydrated' do
  client1 = DemoConnection.new.enable_cache(:api_call).demo_client
￼
  client1.must_be_instance_of DemoConnection::DemoClient
end
```